aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -273,13 +278,24 @@ private boolean isGlueIcebergTable(Table table) {
   public boolean dropTable(TableIdentifier identifier, boolean purge) {
     try {
       TableOperations ops = newTableOps(identifier);
-      TableMetadata lastMetadata = ops.current();
+
+      GlueTableOperations glueOps = (GlueTableOperations) ops;
+      S3FileIO s3FileIO = (S3FileIO) glueOps.io();
+      TableMetadata lastMetadata = null;
+      boolean isTablePurged = isTablePurged(identifier, s3FileIO.client());
+      if (!isTablePurged) {
+        lastMetadata = ops.current();
+      }
+
       glue.deleteTable(DeleteTableRequest.builder()
           .catalogId(awsProperties.glueCatalogId())
           .databaseName(IcebergToGlueConverter.getDatabaseName(identifier))
           .name(identifier.name())
           .build());
       LOG.info("Successfully dropped table {} from Glue", identifier);
+      ValidationException.check(!purge || !awsProperties.glueLakeFormationEnabled() || isTablePurged,
```

Comment:
prefer to make check logic consistent with the error message, in this case `!(purge && awsProperties.glueLakeFormationEnabled() && !isTablePurged)`